eng/targets/ResolveReferences.targets

```diff
@@ -75,6 +80,27 @@
     <Reference Remove="@(Reference)" />
   </ItemGroup>
 
+  <ItemDefinitionGroup>
+    <Reference>
+      <SharedSources>false</SharedSources>
```

Comment:
Can this be conditional on reference name? Or can we define this on `LatestPackageReference` definition level?

These packages are never going to be consumed in any other way so it would be nice to avoid declaring SharedSources on every reference

Reply:
I would love to do this, but I don't know of a simple way in MSBuild to write an expression that does this: `Reference.Where(i => i.Identity.EndsWith(".Sources")).SetMetadata('PrivateAssets', 'All')`. We might be able to do this as an MSBuild task, but every other approach I've tried is expensive computationally.

Reply:
You can set SharedSource metadata in https://github.com/aspnet/AspNetCore/pull/7510/files#diff-51f57d74bf38501ecf1b2a82e10577d7R121. Should be enough for the purposes you are using it for.

Reply:
That's where I would put it, but this isn't valid MSBuild syntax...
```
      <Reference>
        <PrivateAssets Condition="%(Identity.EndsWith('.Sources'))">All</PrivateAssets>
      </Reference>
```
Let me play with some MSBuild intrinsic functions and see if I can find a workaround.